### ISSUE TITLE
Requirements: Keyboard only

### ISSUE THREAD

**Issue by alastc** [w3c/wcag3#267]
Guideline: Can all the functionality be activated by a keyboard interface?

Requirements:
- Keyboard access is enabled
- Keyboard focus is not trapped
- Bi-directional keyboard use is enabled

Links:
- [Google doc](https://docs.google.com/document/d/1uE2WCxPmvNopdCbuZQm_-cGyEdxEouRmZ8UUIlyutoU/edit?tab=t.0#heading=h.4fu2rk4qoiti)

**Comment by patrickhlauke** [comment]
one aspect that WCAG 2.x struggled with (and the discussion is still ongoing) is how to define "standard" keyboard interaction. there is no spec that defines a standard to, say, activate a control. there are de-facto standards, but they're not strong enough to be referenced/enforced in the binary pass/fail 2.x model. in theory then, as long as there's "some" way to control things with a keyboard, they currently nominally pass 2.x.

**Comment by detlevhfischer** [comment]
> no spec that defines a standard to, say, activate a control

One way would be to define minimum standard ways of interacting that must be supported for standard HTML elements like `button`, `a`, `select / option`, and the standard types of `input`? Is there a risk in this? "Minimum" could mean that a `button` or element with `role=button` at least needs to be activatable by ENTER (and hopefully also by space bar). Custom elements could fall under this insofar as the clearly map onto the native ones. That would cover a lot of ground even if components not available natively like `tablist` / `tab` / `tabpanel` might be left out at this point for fear of being too restrictive.

**Comment by patrickhlauke** [comment]
> One way would be to define minimum standard ways of interacting that must be supported for standard HTML elements

where would this be defined though? in the HTML spec? UAAG? it would be weird to define it in WCAG. also, it would need to be defined in a way that works for keyboard-like (but not really keyboard) interfaces.

**Comment by patrickhlauke** [comment]
>  additional layer on top of the default behaviour encoded in native HTML elements 

but that default behaviour is not normatively specified anywhere, so having this in WCAG would effectively be a requirement on user agents? while there might be de-facto default behaviour in most/all browsers, there's no guarantee that even for basic native HTML elements it's consistent with what WCAG would require.

**Comment by scottaohara** [comment]
agree with @patrickhlauke - i could see wcag producing some informative general expectations for common keyboard behavior - similar to how the APG has documented the keyboard commands used for their informative ARIA pattern examples.  But normative keyboard requirements for elements? i dont' see that flying. 

Even with recent work in html to add new features, keyboard behavior is (as always) being left up to the OS platforms.  So even HTML isn't defining explicit requirements for keyboard, nor has it in the past.

**Comment by detlevhfischer** [comment]
> keyboard behavior is (as always) being left up to the OS platforms

OK, I see your point. However, if there is common behavior for the standard elements at least per OS platform, I wonder if it is possible to capture under "Keyboard only" clearly deviating and undocumented kb behavior. It would need to be phrased in a generic way, like "elements can be selected and activated with the keys that commonly work for that element on the respective platform".

**Comment by GreggVan** [comment]
In the keyboard group (now the input group)  we pretty much came to the conclusion that we  (W3C) would need to create and maintain a document with what we think are "standard keyboard navigation commands".   It could list  

- standard keyboard commands supported by most browsers (and a table showing which) 
- suggested future "standard keyboard commands"  and browsers that do 
- other commands that would be "common" but that pages would have to add themselves - that would be helpful

or somethign like that.

then we can say -- if you use these - you are all set.   If you want to use something else you need to tell the user

not final but that was the only path we really saw to talking about "standard" or "common" keyboard commands

**Comment by alastc** [comment]
I'm going to close this as it was an early attempt at tracking requirements. However, we now have a spreadsheet for that, and there is no issue raised.

For folks mentioning standardised keyboard controls, there is a separate informative document for that now. (I'm not sure if it's made it to github yet.)
